src/librustc_target/abi/mod.rs

```diff
@@ -744,6 +749,18 @@ impl FieldsShape {
     }
 }
 
+/// An identifier that specifies the address space that some operation
+/// should operate on. Special address spaces have an effect on code generation,
+/// depending on the target and the address spaces it implements.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AddressSpace(pub u32);
+
+impl Default for AddressSpace {
+    fn default() -> Self {
+        AddressSpace(0)
+    }
+}
```

Comment:
One could derive the `Default` impl: 
```suggestion
#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
pub struct AddressSpace(pub u32);
```

Reply:
TIL the standard primitives also implement `Default`, will change.

Reply:
Fixed in da620823f5fabba1b891217820f7b5db7f2225c2.